### ISSUE TITLE
RNFirebase doesn't compile with use_frameworks!

### ISSUE THREAD

**Issue by paulftw** [invertase/react-native-firebase#252]
I really love using your library but unfortunately iOS compilation breaks when I add `use_frameworks!` to the Podfile.

I was able to combine RNFirebase and [FirebaseUI-iOS](https://github.com/firebase/FirebaseUI-iOS). Phone Authentication in FirebaseUI triggered `firebase.auth().onAuthStateChanged` with correct user Id, which felt great.

However, in order to add "Login with Facebook" I have to enable `use_frameworks!`. Otherwise FirebaseUI won't build - it's a [known issue with FirebaseUI/FBSDKLoginKit combo](https://github.com/firebase/FirebaseUI-iOS/issues/194).
Enabling use_frameworks breaks RNFirebase.

***Quickest way to reproduce*** the issue is to run this script, it will create and build a RN project:
```
cd $(mktemp -d)
pwd > YOU_ARE_HERE.txt
echo FooBar in created on $(date). Working dir >> ~/my_foobars.txt
echo $(pwd) >> ~/my_foobars.txt

open .
react-native init FooBar
cd FooBar
react-native install react-native-firebase
cat >ios/Podfile <<EOF
platform :ios, '9.0'

# comment out the next line to make RNFirebase great again.
use_frameworks!

react_native_path = '../node_modules/react-native'
target 'FooBar' do
  pod 'Yoga', :path => react_native_path + '/ReactCommon/yoga'
  pod 'React', :path => react_native_path, :subspecs => [
    'BatchedBridge',
    'Core',
  ]

  pod 'FirebaseAnalytics'
  pod 'Firebase/Core'
  pod 'Firebase/AdMob'
  pod 'Firebase/Auth'
  pod 'Firebase/Analytics'
  pod 'Firebase/Crash'
  pod 'Firebase/Database'
  pod 'Firebase/DynamicLinks'
  pod 'Firebase/Messaging'
  pod 'Firebase/RemoteConfig'
  pod 'Firebase/Storage'
  pod 'FirebaseUI/Auth'
  pod 'FirebaseUI/Phone'

  pod 'RNFirebase', :path => '../node_modules/react-native-firebase'
end
EOF

cd ios && pod install && cd .. && react-native run-ios
```

### Error:
```
CompileC /var/folders/1_/zbm5sxxj7rbgbvxqv5xdy7pr0000gp/T/tmp.JygzFQ2B/FooBar/ios/build/Build/Intermediates/Pods.build/Debug-iphonesimulator/RNFirebase.build/Objects-normal/x86_64/RNFirebase.o /var/folders/1_/zbm5sxxj7rbgbvxqv5xdy7pr0000gp/T/tmp.JygzFQ2B/FooBar/node_modules/react-native-firebase/ios/RNFirebase/RNFirebase.m normal x86_64 objective-c com.apple.compilers.llvm.clang.1_0.compiler
    cd /var/folders/1_/zbm5sxxj7rbgbvxqv5xdy7pr0000gp/T/tmp.JygzFQ2B/FooBar/ios/Pods
    export LANG=en_US.US-ASCII
    export PATH="/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/usr/bin:/Applications/Xcode.app/Contents/Developer/usr/bin:/Users/paul/google-cloud-sdk/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/opt/X11/bin:/Users/paul/Library/Android/sdk/tools:/Users/paul/.yarn/bin"
    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -x objective-c -arch x86_64 -fmessage-length=0 -fdiagnostics-show-note-include-stack -fmacro-backtrace-limit=0 -std=gnu99 -fobjc-arc -fmodules -gmodules -fmodules-cache-path=/var/folders/1_/zbm5sxxj7rbgbvxqv5xdy7pr0000gp/T/tmp.JygzFQ2B/FooBar/ios/build/ModuleCache -fmodules-prune-interval=86400 -fmodules-prune-after=345600 -fbuild-session-file=/var/folders/1_/zbm5sxxj7rbgbvxqv5xdy7pr0000gp/T/tmp.JygzFQ2B/FooBar/ios/build/ModuleCache/Session.modulevalidation -fmodules-validate-once-per-build-session -Wnon-modular-include-in-framework-module -Werror=non-modular-include-in-framework-module -Xclang -fmodule-implementation-of -Xclang RNFirebase -Wno-trigraphs -fpascal-strings -O0 -fno-common -Wno-missing-field-initializers -Wno-missing-prototypes -Wdocumentation -Wunreachable-code -Wno-implicit-atomic-properties -Wno-arc-repeated-use-of-weak -Wduplicate-method-match -Wno-missing-braces -Wparentheses -Wswitch -Wunused-function -Wno-unused-label -Wno-unused-parameter -Wunused-variable -Wunused-value -Wempty-body -Wuninitialized -Wno-unknown-pragmas -Wno-shadow -Wno-four-char-constants -Wno-conversion -Wconstant-conversion -Wint-conversion -Wbool-conversion -Wenum-conversion -Wshorten-64-to-32 -Wpointer-sign -Wno-newline-eof -Wno-selector -Wno-strict-selector-match -Wundeclared-selector -Wno-deprecated-implementations -DPOD_CONFIGURATION_DEBUG=1 -DDEBUG=1 -DCOCOAPODS=1 -DOBJC_OLD_DISPATCH_PROTOTYPES=0 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator10.3.sdk -fasm-blocks -fstrict-aliasing -Wprotocol -Wdeprecated-declarations -mios-simulator-version-min=8.0 -g -Wno-sign-conversion -Winfinite-recursion -fobjc-abi-version=2 -fobjc-legacy-dispatch -iquote /var/folders/1_/zbm5sxxj7rbgbvxqv5xdy7pr0000gp/T/tmp.JygzFQ2B/FooBar/ios/build/Build/Intermediates/Pods.build/Debug-iphonesimulator/RNFirebase.build/RNFirebase-generated-files.hmap -I/var/folders/1_/zbm5sxxj7rbgbvxqv5xdy7pr0000gp/T/tmp.JygzFQ2B/FooBar/ios/build/Build/Intermediates/Pods.build/Debug-iphonesimulator/RNFirebase.build/RNFirebase-own-target-headers.hmap -I/var/folders/1_/zbm5sxxj7rbgbvxqv5xdy7pr0000gp/T/tmp.JygzFQ2B/FooBar/ios/build/Build/Intermediates/Pods.build/Debug-iphonesimulator/RNFirebase.build/RNFirebase-all-non-framework-target-headers.hmap -ivfsoverlay /var/folders/1_/zbm5sxxj7rbgbvxqv5xdy7pr0000gp/T/tmp.JygzFQ2B/FooBar/ios/build/Build/Intermediates/Pods.build/all-product-headers.yaml -iquote /var/folders/1_/zbm5sxxj7rbgbvxqv5xdy7pr0000gp/T/tmp.JygzFQ2B/FooBar/ios/build/Build/Intermediates/Pods.build/Debug-iphonesimulator/RNFirebase.build/RNFirebase-project-headers.hmap -I/var/folders/1_/zbm5sxxj7rbgbvxqv5xdy7pr0000gp/T/tmp.JygzFQ2B/FooBar/ios/build/Build/Products/Debug-iphonesimulator/RNFirebase/include -I/var/folders/1_/zbm5sxxj7rbgbvxqv5xdy7pr0000gp/T/tmp.JygzFQ2B/FooBar/ios/Pods/Headers/Private -I/var/folders/1_/zbm5sxxj7rbgbvxqv5xdy7pr0000gp/T/tmp.JygzFQ2B/FooBar/ios/Pods/Headers/Public -I/var/folders/1_/zbm5sxxj7rbgbvxqv5xdy7pr0000gp/T/tmp.JygzFQ2B/FooBar/ios/Pods/Headers/Public/Firebase -I/var/folders/1_/zbm5sxxj7rbgbvxqv5xdy7pr0000gp/T/tmp.JygzFQ2B/FooBar/ios/Pods/Headers/Public/FirebaseAnalytics -I/var/folders/1_/zbm5sxxj7rbgbvxqv5xdy7pr0000gp/T/tmp.JygzFQ2B/FooBar/ios/Pods/Headers/Public/FirebaseAuth -I/var/folders/1_/zbm5sxxj7rbgbvxqv5xdy7pr0000gp/T/tmp.JygzFQ2B/FooBar/ios/Pods/Headers/Public/FirebaseCore -I/var/folders/1_/zbm5sxxj7rbgbvxqv5xdy7pr0000gp/T/tmp.JygzFQ2B/FooBar/ios/Pods/Headers/Public/FirebaseCrash -I/var/folders/1_/zbm5sxxj7rbgbvxqv5xdy7pr0000gp/T/tmp.JygzFQ2B/FooBar/ios/Pods/Headers/Public/FirebaseDatabase -I/var/folders/1_/zbm5sxxj7rbgbvxqv5xdy7pr0000gp/T/tmp.JygzFQ2B/FooBar/ios/Pods/Headers/Public/FirebaseDynamicLinks -I/var/folders/1_/zbm5sxxj7rbgbvxqv5xdy7pr0000gp/T/tmp.JygzFQ2B/FooBar/ios/Pods/Headers/Public/FirebaseInstanceID -I/var/folders/1_/zbm5sxxj7rbgbvxqv5xdy7pr0000gp/T/tmp.JygzFQ2B/FooBar/ios/Pods/Headers/Public/FirebaseMessaging -I/var/folders/1_/zbm5sxxj7rbgbvxqv5xdy7pr0000gp/T/tmp.JygzFQ2B/FooBar/ios/Pods/Headers/Public/FirebaseRemoteConfig -I/var/folders/1_/zbm5sxxj7rbgbvxqv5xdy7pr0000gp/T/tmp.JygzFQ2B/FooBar/ios/Pods/Headers/Public/FirebaseStorage -I/var/folders/1_/zbm5sxxj7rbgbvxqv5xdy7pr0000gp/T/tmp.JygzFQ2B/FooBar/ios/Pods/Headers/Public/FirebaseUI -I/var/folders/1_/zbm5sxxj7rbgbvxqv5xdy7pr0000gp/T/tmp.JygzFQ2B/FooBar/ios/Pods/Headers/Public/Google-Mobile-Ads-SDK -I/var/folders/1_/zbm5sxxj7rbgbvxqv5xdy7pr0000gp/T/tmp.JygzFQ2B/FooBar/ios/build/Build/Intermediates/Pods.build/Debug-iphonesimulator/RNFirebase.build/DerivedSources/x86_64 -I/var/folders/1_/zbm5sxxj7rbgbvxqv5xdy7pr0000gp/T/tmp.JygzFQ2B/FooBar/ios/build/Build/Intermediates/Pods.build/Debug-iphonesimulator/RNFirebase.build/DerivedSources -F/var/folders/1_/zbm5sxxj7rbgbvxqv5xdy7pr0000gp/T/tmp.JygzFQ2B/FooBar/ios/build/Build/Products/Debug-iphonesimulator/RNFirebase -include /var/folders/1_/zbm5sxxj7rbgbvxqv5xdy7pr0000gp/T/tmp.JygzFQ2B/FooBar/ios/Pods/Target\ Support\ Files/RNFirebase/RNFirebase-prefix.pch -MMD -MT dependencies -MF /var/folders/1_/zbm5sxxj7rbgbvxqv5xdy7pr0000gp/T/tmp.JygzFQ2B/FooBar/ios/build/Build/Intermediates/Pods.build/Debug-iphonesimulator/RNFirebase.build/Objects-normal/x86_64/RNFirebase.d --serialize-diagnostics /var/folders/1_/zbm5sxxj7rbgbvxqv5xdy7pr0000gp/T/tmp.JygzFQ2B/FooBar/ios/build/Build/Intermediates/Pods.build/Debug-iphonesimulator/RNFirebase.build/Objects-normal/x86_64/RNFirebase.dia -c /var/folders/1_/zbm5sxxj7rbgbvxqv5xdy7pr0000gp/T/tmp.JygzFQ2B/FooBar/node_modules/react-native-firebase/ios/RNFirebase/RNFirebase.m -o /var/folders/1_/zbm5sxxj7rbgbvxqv5xdy7pr0000gp/T/tmp.JygzFQ2B/FooBar/ios/build/Build/Intermediates/Pods.build/Debug-iphonesimulator/RNFirebase.build/Objects-normal/x86_64/RNFirebase.o
In file included from /var/folders/1_/zbm5sxxj7rbgbvxqv5xdy7pr0000gp/T/tmp.JygzFQ2B/FooBar/node_modules/react-native-firebase/ios/RNFirebase/RNFirebase.m:1:
/var/folders/1_/zbm5sxxj7rbgbvxqv5xdy7pr0000gp/T/tmp.JygzFQ2B/FooBar/node_modules/react-native-firebase/ios/RNFirebase/RNFirebase.h:4:9: fatal error: 'React/RCTEventEmitter.h' file not found
#import <React/RCTEventEmitter.h>
        ^
1 error generated.
```

**`...node_modules/react-native-firebase/ios/RNFirebase/RNFirebase.h:4:9: fatal error: 'React/RCTEventEmitter.h' file not found`**

### Why fix it here
It would be really great to have compatibility between RNFirebase and FirebaseUI.
There's an official Android version of FirebaseUI, and Facebook is a rather popular login method, so unblocking a road towards a proper RNFirebaseUI package would be valuable for the whole community.

Root cause is obviously the over-complicated and ugly XCode build system, but maybe there's a not too hard fix on your side.
I will raise an issue with FirebaseUI-iOS as they seem to be open to supporting non-frameworks build but obviously no guarantees or ETAs there.

### Environment
```
$ uname -a
Darwin Redacted.local 16.6.0 Darwin Kernel Version 16.6.0: Fri Apr 14 16:21:16 PDT 2017; root:xnu-3789.60.24~6/RELEASE_X86_64 x86_64
$ xcodebuild -version
Xcode 8.3.3
Build version 8E3004b
$ pod --version
1.3.0.beta.2

"dependencies": {
	"react": "16.0.0-alpha.12",
	"react-native": "0.46.2",
	"react-native-firebase": "^2.0.3"
},
```

**Comment by Salakar** [comment]
Hey Paul, thanks for the detailed issue.

Could you try the steps outlined [here](https://github.com/facebook/react-native/issues/12042#issuecomment-275025960) and [here](https://github.com/facebook/react-native/issues/12952), especially the parts around adding react to build schema and removing `libReact.a` from `Link Binary With Libraries section` - seems like a common react issue when xcode builds using frameworks.

**Comment by paulftw** [comment]
@Salakar I will try those steps, I've already spent three full days trying everything i could think of, but seems like you've just given me some fresh ideas.

Is anyone else able to reproduce it using the script I've provided?
It'd give us an idea of whether this bug affects only certain XCode / Mac OS versions.

**Comment by thevishnupradeep** [comment]
Hi, Any update on this issue?

**Comment by chrisbianca** [comment]
I've just been investigating this further and whilst I've resolved the issue with RNFirebase not being able to see React imports, this only got me to a bigger issue:

Unfortunately it seems that Firebase can't is fundamentally broken when using the `!use_frameworks` flags and it's being imported by a third party library, i.e. RNFirebase.  This is discussed in some detail here: https://github.com/CocoaPods/CocoaPods/issues/5368 and acknowledged by somebody at Google here: https://github.com/firebase/firebase-ios-sdk/issues/58#issuecomment-312885671

FirebaseUI have got around this issue by actually embedding the Firebase dependencies within their library directly.  This approach doesn't work for us, as it would mean every app using RNFirebase would have every Firebase library added to it whether they're using the functionality or not.  This adds unnecessary bloat and is something we're not keen to do.

I've reached out to a contact at Firebase directly to see what we might be able to do in order to resolve this satisfactorily, but until then, I'm afraid that I don't have a way of working around this issue.

**Comment by Ehesp** [comment]
As there's nothing we're able to do and the docs have been updated I'll close this one. Once we have any further feedback from Firebase we can re-open.

**Comment by simonbengtsson** [comment]
It seems this issue is fixed in cocoa pods v4 https://github.com/firebase/firebase-ios-sdk/issues/58#issuecomment-312885671. Is it time to reopen this issue or are there still blocking things?

**Comment by mauscoelho** [comment]
+1

**Comment by luskin** [comment]
+1

**Comment by nagad814** [comment]
Please open this.. i have apps waiting for this.

**Comment by Ehesp** [comment]
@nagad814 check latest release notes.

**Comment by nagad814** [comment]
I did read through them. i am not clear should i use cocoapods or not. if not how to install RNFB

**Comment by chrisbianca** [comment]
@nagad814 I have just updated the release notes to make it clear that you should still use cocoapods for the Firebase SDK, but not for `react-native-firebase` itself

**Comment by skovhus** [comment]
@Salakar @chrisbianca If I understand this correctly, then it seems like the documentation should be updated then? 

https://github.com/invertase/react-native-firebase-docs/blob/9b6b80476aadf0c1f3a51124b74171e278bba593/docs/installation/ios.md#ios-installation still mention that `use_frameworks!` enabled doesn't work.

**Comment by johnanthonyeletto** [comment]
For anyone still having this problem, I found my solution here: [https://stackoverflow.com/a/53178800](https://stackoverflow.com/a/53178800)

In Xcode 10, I had to switch my workspace settings to "Legacy Build System"

**Comment by mikehardy** [comment]
Another possible workaround, from a different library, is to try adding this to podspecs - fixed a build elsewhere without reverting to legacy build:

`s.static_framework = true`

https://github.com/transistorsoft/react-native-background-geolocation/issues/708#issuecomment-488881314

**Comment by MoOx** [comment]
For anyone looking for a workaround with 5.x &  `use_framework!` this days, add this to your Podfile

```podspec
post_install do |installer|
  # react-native-firebase need some adjustements  with use_frameworks!
  rnfirebase = installer.pods_project.targets.find { |target| target.name == 'RNFirebase' }
  rnfirebase.build_configurations.each do |config|
    # react-native-firebase HEADER_SEARCH_PATHS are not good
    config.build_settings['HEADER_SEARCH_PATHS'] = '$(inherited) ${PODS_ROOT}/Headers/Public/**'
    # injecting `s.static_framework = true`
    # (should be in RNFirebase.podspec but since it's not, the code below is doing the same behavior)
    config.build_settings['MACH_O_TYPE'] = 'staticlib'
  end
end
```

**Comment by mikehardy** [comment]
@Salakar this might be interesting - @MoOx - this is fully working for you?

I'm interested for a couple reasons - one because if it doesn't cost too much in maintenance more flexibility is better, but also because your header search paths thing might be the fix I need for #2163  that I'm currently working around with the brutal hack of a header copy in postinstall...

**Comment by mikehardy** [comment]
@MoOx I examined this with @Salakar and the underlying issue should be resolved in v6 but for v5 if you'd like to propose a PR to the actual podspec we could get this done and released. I'm going to examine at least the header paths issue for the above mentioned #2163

**Comment by MoOx** [comment]
@mikehardy PR made that allows me to cleanup my podfile! (not more tricks needed) https://github.com/invertase/react-native-firebase/pull/2235

**Comment by WillyRamirez** [comment]
@MoOx  I tried your script but get the following error:

`[!] An error occurred while processing the post-install hook of the Podfile.

undefined method `build_configurations' for nil:NilClass

/Users/***/projects/***/ios/Podfile:35:in `block (2 levels) in from_ruby'
/usr/local/lib/ruby/gems/2.5.0/gems/cocoapods-core-1.7.2/lib/cocoapods-core/podfile.rb:179:in `post_install!'
/usr/local/lib/ruby/gems/2.5.0/gems/cocoapods-1.7.2/lib/cocoapods/installer.rb:836:in `run_podfile_post_install_hook'
/usr/local/lib/ruby/gems/2.5.0/gems/cocoapods-1.7.2/lib/cocoapods/installer.rb:824:in `block in run_podfile_post_install_hooks'
/usr/local/lib/ruby/gems/2.5.0/gems/cocoapods-1.7.2/lib/cocoapods/user_interface.rb:145:in `message'
/usr/local/lib/ruby/gems/2.5.0/gems/cocoapods-1.7.2/lib/cocoapods/installer.rb:823:in `run_podfile_post_install_hooks'
/usr/local/lib/ruby/gems/2.5.0/gems/cocoapods-1.7.2/lib/cocoapods/installer.rb:309:in `block in create_and_save_projects'
/usr/local/lib/ruby/gems/2.5.0/gems/cocoapods-1.7.2/lib/cocoapods/user_interface.rb:64:in `section'
/usr/local/lib/ruby/gems/2.5.0/gems/cocoapods-1.7.2/lib/cocoapods/installer.rb:290:in `create_and_save_projects'
/usr/local/lib/ruby/gems/2.5.0/gems/cocoapods-1.7.2/lib/cocoapods/installer.rb:281:in `generate_pods_project'
/usr/local/lib/ruby/gems/2.5.0/gems/cocoapods-1.7.2/lib/cocoapods/installer.rb:159:in `install!'
/usr/local/lib/ruby/gems/2.5.0/gems/cocoapods-1.7.2/lib/cocoapods/command/install.rb:51:in `run'
/usr/local/lib/ruby/gems/2.5.0/gems/claide-1.0.2/lib/claide/command.rb:334:in `run'
/usr/local/lib/ruby/gems/2.5.0/gems/cocoapods-1.7.2/lib/cocoapods/command.rb:52:in `run'
/usr/local/lib/ruby/gems/2.5.0/gems/cocoapods-1.7.2/bin/pod:55:in `<top (required)>'
/usr/local/bin/pod:23:in `load'
/usr/local/bin/pod:23:in `<main>'`

**Comment by mikehardy** [comment]
released 5.5.4 with use_frameworks support today!

**Comment by huanvu** [comment]
`error Failed to build iOS project. We ran "xcodebuild" command but it exited with error code 65. To debug build logs further, consider building your app with Xcode.app, by opening hotlap.xcworkspace

2019-07-03 02:10:40.251 xcodebuild[90954:11082035]  DTDeviceKit: deviceType from 6834c928b75d94cf9741389e8b65c368f2be1b44 was NULL
2019-07-03 02:10:40.301 xcodebuild[90954:11082034]  DTDeviceKit: deviceType from 6834c928b75d94cf9741389e8b65c368f2be1b44 was NULL
** BUILD FAILED **


The following commands produced analyzer issues:
        Analyze /Users/huanvu/Documents/hotlap-react/node_modules/react-native/ReactCommon/jsi/jsi.cpp normal x86_64
        Analyze /Users/huanvu/Documents/hotlap-react/node_modules/react-native/ReactCommon/yoga/yoga/Yoga.cpp normal x86_64
        Analyze /Users/huanvu/Documents/hotlap-react/node_modules/react-native/React/Base/RCTModuleMethod.mm normal x86_64
(3 commands with analyzer issues)

The following build commands failed:
        CompileC /Users/huanvu/Documents/hotlap-react/ios/build/hotlap/Build/Intermediates.noindex/RNFirebase.build/developmentDebug-iphonesimulator/RNFirebase.build/Objects-normal/x86_64/RNFirebaseFunctions.o /Users/huanvu/Documents/hotlap-react/node_modules/react-native-firebase/ios/RNFirebase/functions/RNFirebaseFunctions.m normal x86_64 objective-c com.apple.compilers.llvm.clang.1_0.compiler
(1 failure)`

@mikehardy I'm having trouble with 5.5.4 (with 6.3.0 firebase pods) still. Any clue what this could be about?

**Comment by mikehardy** [comment]
Whenever I have an iOS build problem, before proceeding further, I do a deep cleaning. Can you try `npm i —save-dev react-native-clean-project && npx react-native clean-project-auto`, or (even better since it fixes an iOS thing where DerivedData still wasn’t deleted) https://github.com/pmadruga/react-native-clean-project/pull/19 - `npm i —save-dev ‘git+https://github.com/mikehardy/react-native-clean-project.git#xcode-caches' && npx react-native clean-project-auto` ?

**Comment by huanvu** [comment]
@mikehardy deep cleaning with those scripts did not work for me.

I was able to successfully build by unlinking react-native-firebase and adding the pod for RNFirebase to my Podfile like here: https://github.com/invertase/react-native-firebase/pull/2235#issuecomment-507961261

... but the app crashed on launch with not much information that I could decipher because xcode. Will keep digging.

**Comment by mikehardy** [comment]
Interesting - I think for use_frameworks you would need to use react-native-firebase integrated by pods, so that makes sense (and we should mention that in the docs I think, but first we need to know it works - use_frameworks support is new, and it actually breaks again in RN0.60 because of an upstream react-native issue but that’s another story

**Comment by aliustaoglu** [comment]
I use below script to automate https://github.com/invertase/react-native-firebase/issues/252#issuecomment-525430121 by editing local podspecs. But, unfortunately you need to run `pod install` twice for this to work properly. It doesn't look like pre_install is actually applied prior to install.

```ruby
pre_install do |installer|
  puts('Editing RNFirebase podspecs to support use_frameworks')
  installer.sandbox.development_pods.each do |devpod|
    pod_name = devpod[0]
    pod_path = devpod[1]
    if (pod_name.start_with?('RNFB'))
      text = File.read(pod_path)
      new_contents = text.gsub(/  s.static_framework    = false/, "  s.static_framework    = true")
      File.open(pod_path, "w") {|file| file.puts new_contents }
    end
  end
end
```

**Comment by HosseinRashno** [comment]
I can't believe that I spent two days fixing this, while [the solution](https://rnfirebase.io/#allow-ios-static-frameworks) was already there in the documentation!!!!  
  
From the documentation:   
To enable Static Framework support, add the following global to the top of your `/ios/Podfile` file:
```
$RNFirebaseAsStaticFramework = true
```
